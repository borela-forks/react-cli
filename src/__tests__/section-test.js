// @flow strict

import * as React from "react";
import ReactCLI from "../index";

test("sections should be able to render horizontally", done => {
  ReactCLI.render(
    <div horizontal>
      Column 1 <br /> Column 2
    </div>,
    undefined,
    50,
    outputString => {
      expect(outputString).toMatchSnapshot();
      done();
    },
    "+"
  );
});

test("sections should be able to render vertically", done => {
  ReactCLI.render(
    <div>
      Row 1 <br /> Row 2
    </div>,
    undefined,
    50,
    outputString => {
      expect(outputString).toMatchSnapshot();
      done();
    },
    "+"
  );
});

test("sections should be able to align text left", done => {
  ReactCLI.render(
    <div align="left">Left text</div>,
    undefined,
    50,
    outputString => {
      expect(outputString).toMatchSnapshot();
      done();
    },
    "+"
  );
});

test("sections should be able to align text right", done => {
  ReactCLI.render(
    <div align="right">Right text</div>,
    undefined,
    50,
    outputString => {
      expect(outputString).toMatchSnapshot();
      done();
    },
    "+"
  );
});

test("sections should be able to align text center", done => {
  ReactCLI.render(
    <div align="center">Center text</div>,
    undefined,
    50,
    outputString => {
      expect(outputString).toMatchSnapshot();
      done();
    },
    "+"
  );
});
