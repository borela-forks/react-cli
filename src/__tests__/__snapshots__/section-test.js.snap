// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sections should be able to align text center 1`] = `
"+++++++++++++++++++Center text++++++++++++++++++++
"
`;

exports[`sections should be able to align text left 1`] = `
"Left text+++++++++++++++++++++++++++++++++++++++++
"
`;

exports[`sections should be able to align text right 1`] = `
"++++++++++++++++++++++++++++++++++++++++Right text
"
`;

exports[`sections should be able to render horizontally 1`] = `
"Column 1+++++++++++++++++Column 2+++++++++++++++++
"
`;

exports[`sections should be able to render vertically 1`] = `
"Row 1+++++++++++++++++++++++++++++++++++++++++++++
Row 2+++++++++++++++++++++++++++++++++++++++++++++
"
`;
